### PR TITLE
[Modules] MachineLearningWorkspace: Updated Customer-Managed-Key implementation

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated CMK implementation to common pattern
- This also allows us to avoid the need for specifying a particular key version in the parameter file

| Pipeline |
| - |
| [MachineLearningServices: Workspaces](https://github.com/Azure/ResourceModules/actions/workflows/ms.machinelearningservices.workspaces.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.MachineLearningServices/workspaces/deploy.bicep [hunk 1]
```diff
@@ -113,14 +113,17 @@ param description string = ''
...
-param encryptionKeyIdentifier string = ''
+@sys.description('Optional. The name of the customer managed key to use for encryption.')
+param cMKKeyName string = ''
 
-@sys.description('Conditional. The ResourceID of the keyVault where the customer owned encryption key is present. Required if an \'encryptionIdentity\' was provided.')
-param encryptionKeyVaultResourceId string = ''
+@sys.description('Optional. User assigned identity to use when fetching the customer managed key. If not provided, a system-assigned identity can be used - but must be given access to the referenced key vault first.')
+param cMKUserAssignedIdentityResourceId string = ''
+
+@sys.description('Optional. The version of the customer managed key to reference for encryption. If not provided, the latest key version is used.')
```
Comment (eriqua) [inline modules/Microsoft.MachineLearningServices/workspaces/deploy.bicep:125]: I'd suggest to move this parameter up to stay closer to the other `cMKKey` parameters

File: modules/Microsoft.MachineLearningServices/workspaces/.test/parameters.json [hunk 2]
```diff
@@ -34,14 +34,14 @@
         "discoveryUrl": {
             "value": "http://example.com"
         },
-        "encryptionIdentity": {
+        "cMKUserAssignedIdentityResourceId": {
```
Comment (eriqua) [inline modules/Microsoft.MachineLearningServices/workspaces/.test/parameters.json:37]: For other resources we provide a dedicated test through encr.parameters.json. Do we want to keep the same for consistency?

Comment (MrMCake) [reply modules/Microsoft.MachineLearningServices/workspaces/.test/parameters.json:37]: We can. For some resources the test actually needed to be a different file, but not for all. Will adjust

File: modules/Microsoft.MachineLearningServices/workspaces/deploy.bicep [hunk 3]
```diff
@@ -119,12 +119,12 @@ param cMKKeyVaultResourceId string = ''
 @sys.description('Optional. The name of the customer managed key to use for encryption.')
 param cMKKeyName string = ''
 
-@sys.description('Optional. User assigned identity to use when fetching the customer managed key. If not provided, a system-assigned identity can be used - but must be given access to the referenced key vault first.')
-param cMKUserAssignedIdentityResourceId string = ''
-
 @sys.description('Optional. The version of the customer managed key to reference for encryption. If not provided, the latest key version is used.')
 param cMKKeyVersion string = ''
 
+@sys.description('Optional. User assigned identity to use when fetching the customer managed key. If not provided, a system-assigned identity can be used - but must be given access to the referenced key vault first.')
```
Comment (eriqua) [inline modules/Microsoft.MachineLearningServices/workspaces/deploy.bicep:125]: Does this implementation support a system-assigned identity?

Comment (MrMCake) [reply modules/Microsoft.MachineLearningServices/workspaces/deploy.bicep:125]: It appears so. I can re-test it some time once I can

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**